compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -85,39 +89,92 @@ chooseDataTiledMMAAttr(TypeRange eTypes, IREE::GPU::TargetAttr target,
   // Step 2: Select the unrolling factors for the generic case where there is no
   //         narrow dimension.
   //
-  // These hardcoded constants should become functions querying `target`.
-  //
-  // Target ISA preferred load instruction size, in bits.
-  const int kLoadInstructionBits = 128;
-  // Target ISA preferred number of subgroups per block to get full utilization.
-  const int kNumSubgroups = 4;
-  // Number of register space bits to use for accumulators. Should typically be
-  // between 50% and 80% of total available register space, as the accumulator
-  // tends to be larger than the A and B matrix tiles.
-  const int kMaxAccumulatorRegisterBits = 256 * 32;
+  auto wgp = target.getWgp();
+  if (!wgp.getMaxLoadInstructionBits() || !wgp.getVgprSpaceBits() ||
+      !wgp.getWorkgroupSimds()) {
+    // Missing worgroup parameters: data tiling not supported on this target.
+    return std::nullopt;
+  }
 
   MMAAttr intrinsicMma = MMAAttr::get(ctx, *intrinsic);
   auto [intrinsicA, intrinsicB, intrinsicC] = intrinsicMma.getABCVectorTypes();
   // The unrollK factor serves to allow loads from the A and B matrices to use
   // the target ISA's vector loads. For instance, if the ISA has 128-bit loads
   // and each intrinsic consumes only 32 bits from A and B, then we want to set
   // unrollK=4 to turn 4 separate 32-bit loads into one 128-bit load.
-  const int unrollK =
-      kLoadInstructionBits /
-      std::min(
-          intrinsicA.getElementTypeBitWidth() * intrinsicA.getNumElements(),
-          intrinsicB.getElementTypeBitWidth() * intrinsicB.getNumElements());
+  const int unrollK = std::max(
+      1, static_cast<int>(*wgp.getMaxLoadInstructionBits() /
+                          std::min(intrinsicA.getElementTypeBitWidth() *
+                                       intrinsicA.getNumElements(),
+                                   intrinsicB.getElementTypeBitWidth() *
+                                       intrinsicB.getNumElements())));
```

Comment:
A couple of suggestion, because this statement is very long.

1. We can declare `numATotalBits` and `numBTotalBits` which helps the readability a bit.
2. you can write `std::max<int>()`, then you don't need the `static_cast` cast.

Reply:
Re 2., recently @kuhar made the opposite review comment :stuck_out_tongue_closed_eyes:  https://github.com/iree-org/iree/pull/18622/files/7a5c24a0d90e0a7f785d6caf0feb6076bf1d62d6#r1788252470